mempool/cat/pool.go

```diff
@@ -0,0 +1,780 @@
+package cat
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/creachadair/taskgroup"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/mempool"
+	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/types"
+)
+
+// enforce compile-time satisfaction of the Mempool interface
+var _ mempool.Mempool = (*TxPool)(nil)
+
+// Amount of evicted txs to cache. This is low enough so that in patches
+// of high tx throughput, the txpool will be able to quickly recover txs
+// that evicted but not large enough that continual excessive tx load won't
+// take up needless memory.
+const evictedTxCacheSize = 200
+
+var (
+	ErrTxInMempool       = errors.New("tx already exists in mempool")
+	ErrTxAlreadyRejected = errors.New("tx was previously rejected")
+)
+
+// TxPoolOption sets an optional parameter on the TxPool.
+type TxPoolOption func(*TxPool)
+
+// TxPool implemements the Mempool interface and allows the application to
+// set priority values on transactions in the CheckTx response. When selecting
+// transactions to include in a block, higher-priority transactions are chosen
+// first.  When evicting transactions from the mempool for size constraints,
+// lower-priority transactions are evicted sooner.
+//
+// Within the txpool, transactions are ordered by time of arrival, and are
+// gossiped to the rest of the network based on that order (gossip order does
```

Comment:
I think this comment still needs to be updated based on the conversation on the `txToBeBroadcast` map

Reply:
Yup, I've updated it